quarkus-integration/quarkus-benchmark/deployment/src/main/java/ai/timefold/solver/benchmark/quarkus/deployment/TimefoldBenchmarkProcessor.java

```diff
@@ -66,11 +63,35 @@ void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBe
         } else {
             benchmarkConfig = null;
         }
-        syntheticBeans.produce(SyntheticBeanBuildItem.configure(PlannerBenchmarkConfig.class)
-                .supplier(recorder.benchmarkConfigSupplier(benchmarkConfig))
-                .done());
         additionalBeans.produce(new AdditionalBeanBuildItem(TimefoldBenchmarkBeanProvider.class));
         unremovableBeans.produce(UnremovableBeanBuildItem.beanTypes(TimefoldBenchmarkRuntimeConfig.class));
         unremovableBeans.produce(UnremovableBeanBuildItem.beanTypes(SolverConfig.class));
+        return new BenchmarkConfigBuildItem(benchmarkConfig);
+    }
+
+    /**
+     * The build step executes at runtime to fetch an updated instance of properties from
+     * {@link ai.timefold.solver.quarkus.config.TimefoldRuntimeConfig}.
+     * <p>
+     * The reason we need to register the managed beans at runtime is because {@code Arc.container().instance()} does
+     * not return an instance of {@link ai.timefold.solver.quarkus.config.TimefoldRuntimeConfig} when using interfaces
+     * instead of classes. Defining configuration properties as interfaces is the only way to use {@code @WithUnnamedKey}.
+     * This is the default approach documented in both Quarkus and Smallrye pages.
+     * <p>
+     * Finally, recording the bean at runtime is necessary to use updated instances of configuration properties, and
+     * annotating {@link ai.timefold.solver.quarkus.config.TimefoldRuntimeConfig} with {@code @StaticInitSafe} has no effect.
+     */
```

Comment:
This seems like a defensive comment. You're justifying why you did it like this, as if someone was going to say you've done it wrong.

In my opinion, after this code is merged, this comment no longer has any value. We should remove it.

Reply:
The goal was to document the design, but I am fine with discarding it.